.github/workflows/format.yaml

```diff
@@ -7,15 +7,22 @@ on:
   pull_request:
     branches: ["main"]
 
+permissions:
```

Comment:
maybe here we set

```suggestion
permissions: {}
```

and set the permission in the job level and not in the workflow level